chrisj117/CS-472-GROUP-2-PROJECT
# Frontend Registration Setup

**_WHAT DOES THIS PR DO?_**

  - Pushes data using registration form
  - Responds with errors based on server response to registration
  - Pushes data using login form
  - Responds with errors based on server response to login
  - Creates modal popup with information about verification
  - Adds loading spinner when waiting for asynchronous actions
  - Updates navbar based on authentication context
  - Allows navbar on 404 page to control light/dark mode
  - Adds 2 dependencies for the modal and spinner
  
**_WHAT ISSUES ARE RELATED TO THIS PR?_**

  - #99
  - #100 
  - #103 
  - #101 
  - #131

**_HOW DO I TEST OUT THIS PR?_**

  - Make sure the local server is running using `python3 manage.py runserver` in the `api` directory. Head over to the `web` directory and run `npm i` and `npm run dev` to run the frontend. You can test the registration by going to login, clicking "Create an account" or just go to `registration` page in the url. Then you can enter a valid email, username, and password where a verification email will be sent to you to verify.
